File: c-sharp/Battleship.AI/Strategy/NaiveBattleshipStrategy.cs
using Battleship.AI.AITester;

namespace Battleship.AI.Strategy
{
    public class NaiveBattleshipStrategy : IBattleshipStrategy
    {
        private int targetX;
        private int targetY;
        private const int MaxColumns = 10;

        public NaiveBattleshipStrategy()
        {

        }

        public (int x, int y) GetNextMove(FireResult lastResult)
        {
            if (lastResult == FireResult.None)
            {
                targetX = 0;
                targetY = 0;
            }
            else
            {
                targetX++;
                if (targetX >= MaxColumns)
                {
                    targetX = 0;
                    targetY++;
                }
            }

            return (targetX, targetY);
        }
    }
}

File: javascript/src/strategy/naive-battleship-strategy.ts
import {FireResult} from '../fire-result'
import {BattleshipStrategy} from './battleship-strategy'
import {Point} from '../point'

const maxColumns = 10

export class NaiveBattleshipStrategy implements BattleshipStrategy {
  lastPoint: Point

  constructor() {
    this.lastPoint = {x: -1, y: 0}
  }

  nextMove(_: FireResult): Point {
    this.lastPoint =  {x: this.lastPoint.x + 1, y: this.lastPoint.y}
    if (this.lastPoint.x >= maxColumns) {
      this.lastPoint = {x: 0, y: this.lastPoint.y + 1}
    }
    return this.lastPoint
  }
}
